src/main/java/net/snowflake/client/core/arrow/MapConverter.java

```diff
@@ -28,6 +31,30 @@ public Object toObject(int index) throws SFException {
 
   @Override
   public String toString(int index) throws SFException {
-    return vector.getObject(index).toString();
+    ArrowObjectStringRepresentationBuilder builder = new ArrowObjectStringRepresentationBuilder();
+
+    FieldVector vectorUnpacked = vector.getChildrenFromFields().get(0);
+
+    FieldVector keys = vectorUnpacked.getChildrenFromFields().get(0);
+    FieldVector values = vectorUnpacked.getChildrenFromFields().get(1);
```

Comment:
Should we verify here that the children set contains key-children and value-children?

Reply:
I belive this should always work for map vector, but I'll verify for empty one

Reply:
Ok, so there might exist an object of MapVector class that does not have these children, but it seems to be a very weird case. We could either try and verify that it won't happen here (which is probably the case), or simply add a check just to be extra safe.

Reply:
After verification, this shouldn't be empty if used properly, so we are good